### PR TITLE
Make tests resilient against transforms added by plugins

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
During https://github.com/WordPress/gutenberg/pull/14625 I noticed that if a plugin adds a transform our transforms test case fails.
We have an end 2 end test that tests the editor with plugins installed, so the transforms added by the plugins should be ignored.

## How has this been tested?
Verify the end to end tests with plugins installed pass.

### CODE REVIEW

**Comment by gziolo** [comment]
> We have an end 2 end test that tests the editor with plugins installed, so the transforms added by the plugins should be ignored.

What if the plugin would remove a transform from the list or unregister one of the blocks? What if plugins adds an additional transform between core blocks? This might get very tricky and have an impact on other tests as well. I don't know what would be the best solution here. Do you have other ideas? Maybe we should run this particular test suite only when external plugins are disabled?

**Comment by gziolo** [comment]
See my comment https://github.com/WordPress/gutenberg/pull/14625#discussion_r269975473. This might be another way of solving the same issue.

**Comment by jorgefilipecosta** [comment]
> See my comment #14625 (comment). This might be another way of solving the same issue.

Hi @gziolo, after thinking about this problem I preferred to follow your suggestion of not executing the tests when plugins are enabled. Plugins can change the markup of the blocks, change the way a transform works and other actions that would make the tests fail. We don't know the way a plugin changes a transform so the best long term solution is to leave the transform testing as a responsibility of the plugins.
So this PR was updated to not execute transform tests when popular plugins are enabled.

File: packages/e2e-tests/specs/block-transforms.test.js [hunk 1]
```diff
@@ -96,91 +97,101 @@ const getTransformResult = async ( blockContent, transformName ) => {
...
-						distinationBlock,
+						destinationBlock,
 					] )
 				)
 			)
 		);
 
+		const testMessageFormat = 'block %s in fixture %s into the %s block';
+
+		if ( process.env.POPULAR_PLUGINS ) {
```
Comment (gziolo) [inline packages/e2e-tests/specs/block-transforms.test.js]: This test is hard to follow at the moment. 

Is it possible to divide this test suite into 2 groups? One that runs always and the other one which runs only when `POPULAR_PLUGINS` is disabled?

Comment (jorgefilipecosta) [reply packages/e2e-tests/specs/block-transforms.test.js]: Thank you for the suggestion the code was updated and simplified to follow it :)

**Review by gziolo — APPROVED** [review]
It ended up easier than I could anticipate. Great job @jorgefilipecosta  :)

:shipit:
